2024-04-21 04:40:24,152: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-04-21 04:40:27,016: root: INFO: Current backtesting datetime 2023-12-15 08:30:00-05:00
2024-04-21 04:40:27,017: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-04-21 04:40:27,018: root: INFO: Current backtesting datetime 2023-12-15 09:30:00-05:00
2024-04-21 04:40:27,019: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-04-21 04:40:27,020: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-04-21 04:40:27[0m
2024-04-21 04:40:27,090: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-04-21 04:40:27,099: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-04-21 04:40:27,100: root: ERROR: 'datetime.datetime' object has no attribute 'strfttime'
2024-04-21 04:40:27,105: root: ERROR: Traceback (most recent call last):
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 829, in run
    self._run_trading_session()
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 780, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 274, in func_output
    result = func_input(self, *args, **kwargs)
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 298, in func_output
    result = func_input(self, *args, **kwargs)
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 87, in func_output
    return func_input(self, *args, **kwargs)
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\strategies\strategy_executor.py", line 378, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
  File "C:\Users\kushw\miniconda3\envs\tgpu\lib\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
  File "c:\Users\kushw\PROJECTS\python projects\trade\Trading\tradingbot.py", line 57, in on_trading_iteration
    news = self.get_news()
  File "c:\Users\kushw\PROJECTS\python projects\trade\Trading\tradingbot.py", line 41, in get_news
    today,three_days_prior=self.get_dates()
  File "c:\Users\kushw\PROJECTS\python projects\trade\Trading\tradingbot.py", line 38, in get_dates
    return today.strfttime('%Y-%m-$d'),three_days_prior.strfttime('%Y-%m-$d')
AttributeError: 'datetime.datetime' object has no attribute 'strfttime'. Did you mean: 'strftime'?

2024-04-21 04:40:27,109: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-04-21 04:40:27,126: root: INFO: Backtesting finished
2024-04-21 04:40:27,128: root: WARNING: Cannot plot returns because the strategy returns are missing
2024-04-21 04:40:27,389: root: WARNING: Cannot create a tearsheet because the strategy returns are missing
2024-04-21 04:40:27,389: backtest_stats: INFO: Backtest took 0:00:03.243041 for a speed of 0.000
